[CI] UpgradeClusterClientYamlTestSuiteIT.test {p0=old_cluster/10_basic/Component and composable template validation}

**Build scan**:
https://gradle-enterprise.elastic.co/s/gihdwh5jshtqc

**Repro line**:
> ./gradlew ':qa:rolling-upgrade:v8.0.0#oldClusterTest' --tests "org.elasticsearch.upgrades.UpgradeClusterClientYamlTestSuiteIT.test {p0=old_cluster/10_basic/Component and composable template validation}" -Dtests.seed=1CFA140C303404BF -Dtests.security.manager=true -Dtests.locale=sr-CS -Dtests.timezone=Etc/GMT-6 -Druntime.java=11 -Dtests.rest.suite=old_cluster
>

**Reproduces locally?**:
**Yes**

**Applicable branches**:
`master`

**Failure history**:
This is a new failure since the test is just added (#58867). The failure is also seed specific. So I am _not_ muting it just yet.

**Failure excerpt**:
```
java.lang.AssertionError: Failure at [old_cluster/10_basic:228]: got unexpected warning header [ |  
-- | --
  | 299 Elasticsearch-8.0.0-SNAPSHOT-2a0010e5ec92584ab2d44efe97b2d28ff85d95be "index template [my-it] has index patterns [test-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [my-it] will take precedence during new index creation" |  
  | ]
```

--- a/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/old_cluster/10_basic.yml
+++ b/qa/rolling-upgrade/src/test/resources/rest-api-spec/test/old_cluster/10_basic.yml
@@ -207,6 +207,9 @@
 
 ---
 "Component and composable template validation":
+  - skip:
+      features: allowed_warnings
+
   - do:
       cluster.put_component_template:
         name: my-ct
@@ -226,6 +229,8 @@
               eggplant: true
 
   - do:
+      allowed_warnings:
+        - "index template [my-it] has index patterns [test-*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [my-it] will take precedence during new index creation"
       indices.put_index_template:
         name: my-it
         body:

